UnityProject/Assets/Scripts/Messages/Server/UpdateTileMessage.cs

```diff
@@ -10,14 +11,64 @@ public class UpdateTileMessage : ServerMessage
 	public Color Colour;
 	public uint TileChangeManager;
 
+	public static List<delayedData> DelayedStuff = new List<delayedData>();
+
+	public struct delayedData
+	{
+		public Vector3Int Position;
+		public TileType TileType;
+		public string TileName;
+		public Matrix4x4 TransformMatrix;
+		public Color Colour;
+		public uint TileChangeManager;
+
+		public delayedData(Vector3Int inPosition, TileType inTileType, string inTileName, Matrix4x4 inTransformMatrix,
+			Color inColour, uint inTileChangeManager)
+		{
+			Position = inPosition;
+			TileType = inTileType;
+			TileName = inTileName;
+			TransformMatrix = inTransformMatrix;
+			Colour = inColour;
+			TileChangeManager = inTileChangeManager;
+		}
+	}
+
+
 	public override void Process()
 	{
 		LoadNetworkObject(TileChangeManager);
-		var tileChangerManager = NetworkObject.GetComponent<TileChangeManager>();
-		tileChangerManager.InternalUpdateTile(Position, TileType, TileName, TransformMatrix, Colour);
+		if (NetworkObject == null)
+		{
+			DelayedStuff.Add(new delayedData(Position, TileType, TileName, TransformMatrix, Colour, TileChangeManager));
+		}
+		else
+		{
+			var tileChangerManager = NetworkObject.GetComponent<TileChangeManager>();
+			tileChangerManager.InternalUpdateTile(Position, TileType, TileName, TransformMatrix, Colour);
+			TryDoUndone();
+		}
+	}
+
+	public void TryDoUndone()
```

Comment:
the method name is weird

Reply:
What is it doing or undoing?

Reply:
changes that couldn't be applied when it was sent,it's trying to do them again